### ISSUE TITLE
Found rtl8821au 4.3.19 version

### ISSUE THREAD

**Issue by aldegoeij** [diederikdehaas/rtl8812AU#29]
Bough an Edimax AC600 (EW-7811DAC) 2.4 and 5 Ghz dongle which comes with a rtl821au 4.3.19 stamped driver here: http://www.edimax.nl/edimax/download/download/data/edimax/nl/download/for_home/wireless_adapters/wireless_adapters_ac600_dual-band/ew-7811dac

Is this useful to add to your repo? (knowing you are already working on 22-beta)

**Comment by diederikdehaas** [comment]
> Is this useful to add to your repo? (knowing you are already working on 22-beta)

It would, were it not that I had recently found the [4.3.20 driver](http://www.edimax.nl/edimax/download/download/data/edimax/nl/download/for_home/wireless_adapters/wireless_adapters_ac1200_dual-band/ew-7822uac) and I will be adding that to this repo.
Thanks for sharing :+1:

**Comment by uminokoe** [comment]
In the meantime, please note that additional changes are needed for newer kernels. The download page states that the supported kernel versions are 2.6.18 - 4.2. However, the great thing is that the documentation in the .zip file can be quite useful.

**Comment by diederikdehaas** [comment]
Yep, it's quite likely I'll re-apply various changes we've done in the repo to the 4.3.20 driver, but that one does seem to be a good candidate to be promoted as default. Despite the (slightly) lower version number than 4.3.22-beta, the code does seem to be more recent.

**Comment by uminokoe** [comment]
Grawp's repository now contains the 4.3.20 version of the driver (https://github.com/Grawp/rtl8812au_rtl8821au) and made changes to support kernel version >=4.7. It seems to me that, although 4.3.20 fixes some issues, some codes present in 4.3.22-beta have yet to be included in 4.3.20. I hope the 4.3.22-beta branch will be kept.

**Comment by diederikdehaas** [comment]
> I hope the 4.3.22-beta branch will be kept.

It will. But I will change the **default** branch from 4.3.14 to 4.3.20 :wink:

**Comment by uminokoe** [comment]
Using the 4.3.21 driver from EDIMAX (which is originally designed for RTL8814AU), I modified it and it can then be used for RTL8811AU / RTL8812AU / RTL8821AU. You may view the code here: [https://github.com/uminokoe/rtl8812AU/tree/driver-4.3.21](https://github.com/uminokoe/rtl8812AU/tree/driver-4.3.21)
However, please note that it is experimental and I cannot guarantee that everything works. Should you find any bugs, please do not hesitate to submit a pull request or an issue.

**Comment by diederikdehaas** [comment]
> It seems to me that, although 4.3.20 fixes some issues, some codes present in 4.3.22-beta have yet to be included in 4.3.20

Indeed.
I'm working on the 4.3.20 driver and your PRs regarding VHT make it fail as apparently that part of the code didn't make it into 4.3.20, while it is present in 4.3.22-beta and I just saw also in your 4.3.21 driver. :thinking: 

EDIT TO ADD:
Do you have the original driver 'zip' file you used for me? 
I know I can get the _code_ from your repo, but I want to compare your zip with mine and the original date/time stamps could be of use.

**Comment by diederikdehaas** [comment]
I've uploaded my **WIP** branch here: https://github.com/diederikdehaas/rtl8812AU/tree/driver-4.3.20-add-vht-capabilities-wip

**Comment by fedux** [comment]
I found what seems to be the original ZIP for the version @uminokoe is talking about here: http://www.edimax.co.uk/edimax/download/download/data/edimax/uk/download/for_home/wireless_adapters/wireless_adapters_ac1750_dual-band/ew-7833uac

**Comment by uminokoe** [comment]
Thank you @fedux .
@diederikdehaas , I was not aware of your updated comment and I apologize for that. Indeed, I downloaded the driver for EW7833UAC (RTL8814AU chipset), unzipped that and converted all line endings to LF. Then the whole folder was uploaded and became the Initial Import.

**Comment by diederikdehaas** [comment]
No worries :+1: 
Unfortunately GitHub doesn't inform about updates to comments, so it's easily missed.

**Comment by weedy** [comment]
@uminokoe does your wifi stick work with your driver-4.3.21 branch? 
Are you using your driver from everyday?

**Comment by uminokoe** [comment]
@weedy 
Yes, I do use it every day. See [this comment](https://github.com/diederikdehaas/rtl8812AU/issues/10#issuecomment-253115897) for the full reply.

**Comment by diederikdehaas** [comment]
Just as I thought I could close this issue because I uploaded the 4.3.20 driver, I noticed that the 4.3.19 driver is for the rtl88**21**au and not for the rtl88**12**au. _sigh_

I need a break.

(importing the `hal/phydm/rtl8821a` folder into the driver-4.3.20 branch and enabling 8821AU in the `Makefile` works)

**Comment by Goga777** [comment]
> Bough an Edimax AC600 (EW-7811DAC) 2.4 and 5 Ghz dongle which comes with a rtl821au 4.3.19 >stamped driver 

with which drivers could you run the ew-7811dac which based on 8811au chip ?

**Comment by uminokoe** [comment]
You may wish to download the driver for EW-7811DAC from Edimax's site, but the driver may need modification to work on newer kernels. Alternatively, try [diederikdehaas' 4.3.22-beta version](https://github.com/diederikdehaas/rtl8812AU/tree/driver-4.3.22-beta) and it should work well with the RTL8811AU chip.
